zeek

```diff
@@ -18,179 +18,214 @@
 	if [ ! -s "$HOST_ZEEK/etc/node.cfg" ]; then
 		echo "Could not find $HOST_ZEEK/etc/node.cfg. Generating one now." >&2
-
-		$SUDO docker run --rm -it --network host \
-		--mount source="$HOST_ZEEK/etc/",destination=/zeek/,type=bind \
-		"$IMAGE_NAME" \
-		zeekcfg -o /zeek/node.cfg --type afpacket
+		docker exec $container zeekcfg -o "/zeek/etc/node.cfg" --type afpacket --processes 0 --no-pin
 	fi
-
-	# create the volumes required for peristing user-installed zkg packages
-	$SUDO docker volume create zeek-zkg-script >/dev/null
-	$SUDO docker volume create zeek-zkg-plugin >/dev/null
-	$SUDO docker volume create zeek-zkg-state >/dev/null
-
-	docker_cmd=("docker" "run" "--detach")  # start container in the background
-	docker_cmd+=("--name" "$CONTAINER_NAME") # provide a predictable name
-	docker_cmd+=("--restart" "$RESTART_POLICY")
-	docker_cmd+=("--cap-add" "net_raw" "--cap-add" "net_admin") # allow Zeek to listen to raw packets
-	docker_cmd+=("--network" "host") # allow Zeek to monitor host network interfaces
-	# allow packages installed via zkg to persist across restarts
-	docker_cmd+=("--mount" "source=zeek-zkg-script,destination=/usr/local/zeek/share/zeek/site/packages/,type=volume")
-	docker_cmd+=("--mount" "source=zeek-zkg-plugin,destination=/usr/local/zeek/lib/zeek/plugins/packages/,type=volume")
-	docker_cmd+=("--mount" "source=zeek-zkg-state,destination=/root/.zkg,type=volume")
-	# mirror the host timezone settings to the container
-	docker_cmd+=("--mount" "source=/etc/localtime,destination=/etc/localtime,type=bind,readonly")
-	# persist and allow accessing the logs from the host
-	docker_cmd+=("--mount" "source=$HOST_ZEEK/logs,destination=/usr/local/zeek/logs/,type=bind")
-	docker_cmd+=("--mount" "source=$HOST_ZEEK/spool,destination=/usr/local/zeek/spool/,type=bind")
-	# allow users to provide arbitrary custom config files and scripts
-	# mount all zeekctl config files
-	while IFS=  read -r -d $'\0' CONFIG; do
-    	docker_cmd+=("--mount" "source=$CONFIG,destination=/usr/local/zeek/${CONFIG#"$HOST_ZEEK"},type=bind")
-	done < <(find "$HOST_ZEEK/etc/" -type f -print0 2>/dev/null)
-	# mount all zeek scripts
-	while IFS=  read -r -d $'\0' SCRIPT; do
-    	docker_cmd+=("--mount" "source=$SCRIPT,destination=/usr/local/zeek/${SCRIPT#"$HOST_ZEEK"},type=bind")
-	done < <(find "$HOST_ZEEK/share/" -type f -print0 2>/dev/null)
-		# loop reference: https://stackoverflow.com/a/23357277
-		# ${CONFIG#"$HOST_ZEEK"} strips $HOST_ZEEK prefix
-	docker_cmd+=("$IMAGE_NAME")
-
-	echo "Starting the Zeek docker container" >&2
-	$SUDO "${docker_cmd[@]}"
-
-	# Fix current symlink for the host (sleep to give Zeek time to finish starting)
-	(sleep 15s; $SUDO docker exec "$CONTAINER_NAME" ln -sfn "$HOST_ZEEK/spool/manager" /usr/local/zeek/logs/current) &
-
-	;;
-stop)
-	#Command(s) needed to stop the service right now
-
-	if [ "$CONTAINER_RUNNING" == "false" ]; then
-		echo "Zeek is already stopped." >&2
-		exit 0
+}
+
+main() {
```

Comment:
Moved the main portion of this script into a function. This allows the entire script to be sourced and used as a library without side effects.